### PR TITLE
fix: Video freezes on a group video call (AR-2316)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2316" title="AR-2316" target="_blank">BugAR-2316</a>  Video is freezing on Android client
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

In a video group call, some videos becomes freezed when someone join/left the call or someone else turned on his camera

### Causes (Optional)

Every video renderer created should be disconnected from the container(parent view) before creating new ones.

### Solutions

Every time the screen is recomposed due to camera state change, I  disconnect old renderers by calling `removeAllViews() `from parent view 
Removing old renders and creating new ones should be only done when there is a change on camera, that's why I am using `shouldRecomposeVideoRenderer`

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

You can test it after merging the PR for ordering the participants wireapp/kalium#840

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/calling/ParticipantTile.kt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package com.wire.android.ui.calling
 
+import android.util.Log
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/calling/ParticipantTile.kt:3]: I believe this is not used 🤔

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-972/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4411128)(4411128)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4411128)(4411128)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/calling/common/CallingGridView.kt [hunk 2]
```diff
@@ -25,6 +25,8 @@ import com.wire.android.ui.home.conversationslist.model.Membership
 import com.wire.android.ui.theme.wireDimensions
 import com.wire.kalium.logic.data.id.QualifiedID
 
+val lastParticipants = mutableListOf<UICallParticipant>()
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/calling/common/CallingGridView.kt:28]: Why we need to have this variable in the outer scope ? Is it possible to pass it between both functions? 🤔

**Review by yamilmedina — APPROVED** [review]
Great work ! 🥇 ⚽ I leave it approved, and raised a question about a var definition.

**Review by alexandreferris — APPROVED** [review]
👌🏻

File: app/src/main/kotlin/com/wire/android/ui/calling/common/CallingGridView.kt (hunk 2 above)
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/calling/common/CallingGridView.kt:28]: It's global list, shared with different file

File: app/src/main/kotlin/com/wire/android/ui/calling/ParticipantTile.kt (hunk 1 above)
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/calling/ParticipantTile.kt:3]: correct

**Pull request merged**